File: Demo/VLFRunnerObjC/VFLExample.m
//
//  VFLExample.m
//  VLFRunnerObjC
//
//  Created by Sidharth Juyal on 09/07/2023.
//

#import "VFLExample.h"

NSString *VFLExampleGetName(VFLExample example)
{
  switch (example) {
    case VFLExampleFull: return @"Screen Full Layout";
    case VFLExampleHSplit: return @"Horizontal Equal Split";
    case VFLExampleVSplit: return @"Vertical Equal Split";
    case VFLExampleFixed: return @"Fixed Size View";
    case VFLExampleCenter: return @"Center View";
    case VFLExampleComplex: return @"Complex Example";
    case VFLExampleTotal: return @"";
  }
  return @"";
}

NSString *VFLExampleGetClassname(VFLExample example)
{
  switch (example) {
    case VFLExampleFull: return @"VFLExampleFullView";
    case VFLExampleHSplit: return @"VFLExampleHSplitView";
    case VFLExampleVSplit: return @"VFLExampleVSplitView";
    case VFLExampleFixed: return @"VFLExampleFixedView";
    case VFLExampleCenter: return @"VFLExampleCenterView";
    case VFLExampleComplex: return @"VFLExampleComplexView";
    case VFLExampleTotal: return nil;
  }
  return nil;
}

File: Demo/VFLRunner/VFLExample.swift
//
//  VFLExample.swift
//  VFLRunner
//
//  Created by Sidharth Juyal on 31/12/2022.
//

import Foundation

struct VFLExample {
    let name: String
    let viewCount: Int
    let formats: [String]
}

struct VFLExamples {
    static var all: [VFLExample] {
        return [
            VFLExample(
                name: "Screen Full Layout",
                viewCount: 1,
                formats: ["V:|[vw0]|", "H:|[vw0]|"]
            ),
            
            VFLExample(
                name: "Horizontal Equal Split",
                viewCount: 2,
                formats: [
                    "V:|[vw0]|",
                    "V:|[vw1]|",
                    "H:|[vw0(==vw1)][vw1]|"
                ]
            ),
            
            VFLExample(
                name: "Vertical Equal Split",
                viewCount: 2,
                formats: [
                    "V:|[vw0(==vw1)][vw1]|",
                    "H:|[vw0]|",
                    "H:|[vw1]|"
                ]
            ),

            VFLExample(
                name: "Fixed Size View",
                viewCount: 1,
                formats: [
                    "V:|-[vw0(100)]",
                    "H:|-[vw0(100)]"
                ]
            )
        ]
    }
}
